### PR TITLE
Simulate vehicle movements

### PR DESCRIPTION
**Pull request by Ranray**

### CODE REVIEW

File: simulation/src/main/java/nl/earthbird/simulation/SimulationRunner.java [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+            counter++;
+            controller.step();
+        }
+        controller.stop();
+    }
+
+    private void updateVehicleLocation(MessageDigest md) throws SumoConnectionException {
+        for (String vehicle : controller.getActiveVehicles()) {
+            GeoLocation location = controller.getVehicleGeoLocation(vehicle);
+            String vehicle_hash = bytesToHex(md.digest(vehicle.getBytes()));
```
Comment (TehMessiah) [inline simulation/src/main/java/nl/earthbird/simulation/SimulationRunner.java]: MINOR Rename this local variable to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](http://cc.pupper.nl/coding_rules#rule_key=squid%3AS00117)

File: simulation/src/main/java/nl/earthbird/simulation/SimulationRunner.java [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+        int counter = interval;
+
+        controller.start();
+
+        controller.step();
+        controller.step();
+        while (0 < controller.getActiveVehicles().size()) {
+            if (Objects.equals(counter, interval)) {
+                updateVehicleLocation(md);
```
Comment (TehMessiah) [inline simulation/src/main/java/nl/earthbird/simulation/SimulationRunner.java]: MAJOR "NullPointerException" will be thrown when invoking method "updateVehicleLocation()". [rule](http://cc.pupper.nl/coding_rules#rule_key=squid%3AS2259)

File: simulation/src/main/java/nl/earthbird/simulation/SimulationRunner.java [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+
+    private Logger logger = Logger.getLogger(this.getClass().getName());
+
+    private SimulationController controller;
+    private MessageController messageController;
+
+    private int interval;
+
+    private static final String ALGORITHM = "MD5";
+    private final static char[] hexArray = "0123456789ABCDEF".toCharArray();
```
Comment (TehMessiah) [inline simulation/src/main/java/nl/earthbird/simulation/SimulationRunner.java]: MINOR Reorder the modifiers to comply with the Java Language Specification. [rule](http://cc.pupper.nl/coding_rules#rule_key=squid%3AModifiersOrderCheck)

**Comment by TehMessiah** [comment]
SonarQube analysis reported 3 issues
* ![MAJOR][MAJOR] 1 major
* ![MINOR][MINOR] 2 minor

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by TehMessiah** [comment]
SonarQube analysis reported 1 issue
* ![MAJOR][MAJOR] 1 major

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: domain/src/main/java/nl/earthbird/simulation/dto/DetectionDto.java [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+        return serialNumber;
+    }
+
+    public byte[] serialize() {
+        return new GsonBuilder().create().toJson(this).getBytes();
+    }
+
+    @Override
+    public String toString() {
+        return "DetectionDto{" +
```
Comment (Laugslander) [inline domain/src/main/java/nl/earthbird/simulation/dto/DetectionDto.java]: Misschien String.format gebruiken hier?

File: exception/src/main/java/nl/earthbird/exception/SumoSetUpException.java [hunk 5]
```diff
@@ -0,0 +1,8 @@
+package nl.earthbird.exception;
+
+public class SumoSetUpException extends Exception {
+
+    public SumoSetUpException() {
+        super("An error has occurred while setting up a connection to SUMO.");
```
Comment (Laugslander) [inline exception/src/main/java/nl/earthbird/exception/SumoSetUpException.java]: What is the difference with SumoConnectionException (description)? 
I think this exception can be removed

Comment (Ranray) [reply exception/src/main/java/nl/earthbird/exception/SumoSetUpException.java]: Changed the text within the Exception.

**Pull request merged**